src/linagora.esn.calendar/app/components/calendars-list/calendars-list.controller.js

```diff
@@ -31,12 +31,12 @@ const _ = require('lodash');
       self.hiddenCalendars = {};
       self.toggleCalendar = calendarVisibilityService.toggle;
 
-      self.activate();
+      return self.activate();
     }
 
     function activate() {
-      $q
-        .all(listCalendars(), getHiddenCalendars())
+      return Promise
```

Comment:
Is there a reanson why sometimes `$q` needs to be changed to `Promise`, but not always?

Reply:
Sometimes $q creates a promise that cannot be triggered, both normal Js way and the $rootScope.$digest() way.
So to make it safe and easy to predict, I use Promise in some hard cases.

Reply:
Inconsistency can be a problem here. We need to either use only `$q` or only `Promise`, mixing both of them can be confusing. Maybe we can create another issue to handle that problem.